File: lab/c/textos/triangulo_asterisco.c
#include <stdio.h>

// Versão 1
void triangulo_asterisco(int n) {
    for (int i = 1; i <= n; i++) {
        for (int j = 0; j < i; j++) {
            printf("*");
        }
        printf("\n");
    }
}

// Versão 2
void triangulo_asterisco2(int n) {
    for (int i = 1; i <= n; i++) {
        char linha[100];
        int k = 0;
        for (int j = 0; j < i; j++) {
            linha[k++] = '*';
        }
        linha[k] = '\0';
        printf("%s\n", linha);
    }
}

// Versão 3
void triangulo_asterisco3(int n) {
    for (int i = 1; i <= n; i++) {
        for (int j = 0; j < i; j++) {
            printf("*");
        }
        printf("\n");
    }
}

// Versão 4
void triangulo_asterisco4(int n) {
    char linha[100] = "*";
    for (int i = 1; i <= n; i++) {
        printf("%s\n", linha);
        int len = i; // tamanho atual
        linha[len] = '*';
        linha[len + 1] = '\0';
    }
}

int main(void) {
    triangulo_asterisco(5);
    printf("\n");
    triangulo_asterisco2(5);
    printf("\n");
    triangulo_asterisco3(5);
    printf("\n");
    triangulo_asterisco4(5);
    return 0;
}

File: lab/javascript/textos/triangulo_asterisco.js
// Versão 1
function triangulo_asterisco(n) {
  for (let i = 1; i <= n; i++) {
    console.log("*".repeat(i));
  }
}

// Versão 2
function triangulo_asterisco2(n) {
  for (let i = 1; i <= n; i++) {
    let linha = "";
    for (let j = 0; j < i; j++) {
      linha += "*";
    }
    console.log(linha);
  }
}

// Versão 3
function triangulo_asterisco3(n) {
  for (let i = 1; i <= n; i++) {
    let linha = "";
    for (let j = 0; j < i; j++) {
      linha += "*";
    }
    console.log(linha);
  }
}

// Versão 4
function triangulo_asterisco4(n) {
  let x = "*";
  for (let i = 1; i <= n; i++) {
    console.log(x);
    x += "*";
  }
}

triangulo_asterisco(5);
console.log();
triangulo_asterisco2(5);
console.log();
triangulo_asterisco3(5);
console.log();
triangulo_asterisco4(5);
